src/libraries/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -200,18 +195,33 @@ public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSyste
                 throw new ArgumentOutOfRangeException(nameof(bufferSize), SR.ArgumentOutOfRange_NeedPosNum);
             }
 
-            // Do not allow using combinations of non-writing file system rights with writing file modes
+            // Do not combine writing modes with exclusively read rights
+            // Write contains AppendData, WriteAttributes, WriteData and WriteExtendedAttributes
```

Comment:
Here's the code that verifies this in .NET Framework: https://referencesource.microsoft.com/#mscorlib/system/io/filestream.cs,673